mesonbuild/build.py

```diff
@@ -2186,6 +2186,8 @@ def flatten_command(self, cmd):
             elif isinstance(c, (BuildTarget, CustomTarget)):
                 self.dependencies.append(c)
                 final_cmd.append(c)
+            elif isinstance(c, CustomTargetIndex):
```

Comment:
you also need `self.dependencies.append(c.target)` here, or you won't create a proper dependency on the target.

You'll also need to emit a `FeatureNew.single_use()` here, as this is a new feature for 0.60

Reply:
done